fotoobo/helpers/vault.py

```diff
@@ -0,0 +1,186 @@
+makes this module very independent.
+"""
+
+import logging
+from pathlib import Path
+from typing import Any, Optional
+
+import requests
+
+log = logging.getLogger("fotoobo")
+
+
+class Client:  # pylint: disable=too-many-instance-attributes
+    """The approle helper class for the Hashicorp Vault
+
+    This vault client gives you methods for your login with approle and maintaining token and data
+    requests.
+    """
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        url: str,
+        namespace: str,
+        data_path: str,
+        role_id: str,
+        secret_id: str,
+        token_file: Optional[str] = None,
+        token_ttl_limit: int = 0,
+    ) -> None:
+        """Initialize the vault client
+
+        Args:
+            url:                The URL of your vault service (eg. https://vault.local:443)
+            namespace:          The namespace of your vault
+            data_path:          The path where the vault data for fotoobo is stored
+            role_id:            The approle role_id
+            secret_id:          The approle secret_id
+            token_file:         The file to store the access token to. If no file is given the token
+                                is not loaded or stored to a file and every execution will issue a
+                                new token.
+            token_ttl_limit:    Set a token limit. If the ttl of a token falls below this limit we
+                                will automatically issue a new token. Default: 0
+        """
+        log.debug("Initialize the vault client")
+        self.namespace: str = namespace
+        self.data_path: str = data_path
+        self.role_id: str = role_id
+        self.secret_id: str = secret_id
+        self.token: str = ""
+        self.token_file: Optional[Path] = None
+        self.token_ttl_limit: int = token_ttl_limit
+        self.url: str = url.strip("/")
+
+        log.debug("vault_client_url: '%s'", self.url)
+        log.debug("vault_client_namespace: '%s'", self.namespace)
+        log.debug("vault_data_path: '%s'", self.data_path)
+        log.debug("vault_client_role_id: '%s...%s'", self.role_id[:4], self.role_id[-5:-1])
+        log.debug("vault_client_secret_id: '%s...%s'", self.secret_id[:4], self.secret_id[-5:-1])
+        log.debug("vault_client_token_ttl_limit: '%s'", self.token_ttl_limit)
+
+        if token_file:
+            self.token_file = Path(token_file).expanduser()
+            log.debug("vault_client_token_file: '%s'", self.token_file)
+            self.load_token()
+
+    def get_data(self, timeout: int = 3) -> Any:
+        """Get data from the key/value store
+
+        Args:
+            path:       The path to your data in the key/value store
+            timeout:    The time before a request to the vault is cancelled
+
+        Returns:
+            The date from the vault
+        """
+        if not self.token:
+            self.get_token()
```

Comment:
Response from get_token() is not used -> What happens if no token can be got?

Reply:
Added a check and raise an exception if no token could be retreived